addons/delivery/tests/test_delivery_cost.py

```diff
@@ -391,3 +391,53 @@ def test_price_with_weight_volume_variable(self):
             qty * list_price * weight * volume,
             "The shipping price is not correctly computed with variable weight*volume.",
         )
+
+    def test_update_weight_in_shipping_when_change_quantity(self):
+        product_test = self.env['product.product'].create({
+            'name': 'Test product',
+            'weight': 1,
+        })
+        sale_order = self.env['sale.order'].create({
+            'partner_id': self.partner_18.id,
+            'order_line': [
+                (0, 0, {
```

Comment:
```suggestion
                Command.create({
```
Nitpick: [Commands](https://www.odoo.com/documentation/16.0/developer/reference/backend/orm.html#odoo.fields.Command) make the code a bit more readable. :slightly_smiling_face: